### PR TITLE
Update the release checklist

### PR DESCRIPTION
**Pull request by BenjaminBossan**
As discussed, we wanted to make small amendments to the release process, so that we have a 0.N.0 commit on main. I also adjusted the wording here and there.

There were some open questions, which I have marked with a `FIXME`. Please let me know how to proceed with those.

Note: For me, the GitHub diff view is not correct, so please check the changes carefully.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -63,19 +63,27 @@
...
-#    Push the tag to git: git push --tags origin main
-# 4. Run the following commands in the top-level directory:
+# 1. Change the version in __init__.py and setup.py to the release version, e.g. from "0.6.0.dev0" to "0.6.0"
+# 2. Check if there are any deprecations that need to be addressed for this release by seaching for "# TODO" in the code
+# 3. Commit these changes with the message: "Release: VERSION", create a PR and merge it.
+# 4. Add a tag in git to mark the release: "git tag VERSION -m 'Adds tag VERSION for pypi' "
+#    Push the tag to git:
+#      git push --tags origin main
+#    It is necessary to work on the original repository, not on a fork.
+#    FIXME should we create an annotated tag: git tag -a ...
```
Comment (pacman100) [inline setup.py:73]: Yes, let's use annotated tag as it is more informative and better for public releases

File: setup.py [hunk 2]
```diff
@@ -63,19 +63,27 @@
...
+#    FIXME we can also skip this and let GH release create the tag for us
```
Comment (pacman100) [inline setup.py:74]: Let's explicitly have the tag step before release.

File: setup.py [hunk 3]
```diff
@@ -63,19 +63,27 @@
...
 #      python setup.py bdist_wheel
 #      python setup.py sdist
-# 5. Upload the package to the pypi test server first:
+#    Ensure that you are on the clean and up-to-date main branch (git status --untracked-files=no should not list any
+#    files and show the main branch)
+# 6. Upload the package to the pypi test server first:
 #      twine upload dist/* -r pypitest
 #      twine upload dist/* -r pypitest --repository-url=https://test.pypi.org/legacy/
-# 6. Check that you can install it in a virtualenv by running:
+#    FIXME are both commands needed? I think not
```
Comment (pacman100) [inline setup.py:83]: Yes, I think the second one isn't required

**Review by pacman100 — COMMENTED** [review]
Thank you @BenjaminBossan for updating the release checklist! Have replied to the FIXME parts. Good to merge once those are addressed

**Comment by BenjaminBossan** [comment]
@pacman100 I updated the PR according to your comments, please check again.

**Review by younesbelkada — APPROVED** [review]
Thanks!

**Pull request merged**